crates/transaction-pool/src/pool/pending.rs

```diff
@@ -59,6 +59,18 @@ impl<T: TransactionOrdering> PendingPool<T> {
         }
     }
 
+    /// Clear all transactions from the pool without resetting other values.
+    /// Used for atomic reordering during basefee update.
+    ///
+    /// # Returns
+    ///
+    /// Returns all transactions by id.
+    fn clear_transactions(&mut self) -> BTreeMap<TransactionId, Arc<PendingTransaction<T>>> {
+        self.independent_transactions.clear();
+        self.all.clear();
+        std::mem::take(&mut self.by_id)
```

Comment:
this needs a note that it does not update size tracker

Reply:
removed